### ISSUE TITLE
OSX and Tailscale returning from sleep -> no DNS working

### ISSUE THREAD

**Issue by shuther** [tailscale/tailscale#12615]
### What is the issue?

The problem is intermittent. When I wake up my computer, DNS is not working.
It seems there are open issues when using other VPN (not my case) or on iOS; not sure if connected.

Extra notes:
no extra VPN, connection is wireless
Other devices (connected or not through tailscale are working fine).
exit node not used
`Use Tailscale DNS settings` on
`Allow Local network access` on
DNS to tailscale nodes is working

### Steps to reproduce

I wake up my computer, internet is working (I can ping 8.8.8.8, but not google.com)

Turning off Tailscale is solving the problem.
Turning it on again works as expected.

### Are there any recent changes that introduced the issue?

to debug further (while it was not working), I ran:
I am not sure if it is expected or what to run if it happens again (resolver 1 and 2 seem mixed?).

```
scutil --dns
DNS configuration

resolver #1
  search domain[0] : www.ts.net
xxx
  nameserver[0] : 100.100.100.100
  if_index : 31 (utun4)
  flags    : Supplemental, Request A records, Request AAAA records
  reach    : 0x00000003 (Reachable,Transient Connection)
  order    : 101400

resolver #2
  nameserver[0] : 192.168.0.2
  if_index : 12 (en0)
  flags    : Request A records
  reach    : 0x00020002 (Reachable,Directly Reachable Address)
  order    : 200000

resolver #3
  domain   : lan.
  nameserver[0] : 100.100.100.100
  if_index : 31 (utun4)
  flags    : Supplemental, Request A records, Request AAAA records
  reach    : 0x00000003 (Reachable,Transient Connection)
  order    : 101404

resolver #4
  domain   : local.
  nameserver[0] : 100.100.100.100
  if_index : 31 (utun4)
  flags    : Supplemental, Request A records, Request AAAA records
  reach    : 0x00000003 (Reachable,Transient Connection)
  order    : 101403

resolver #5
  domain   : xxx.tailscale.net.
  nameserver[0] : 100.100.100.100
  if_index : 31 (utun4)
  flags    : Supplemental, Request A records, Request AAAA records
  reach    : 0x00000003 (Reachable,Transient Connection)
  order    : 101402

resolver #6
  domain   : pxxx.ts.net.
  nameserver[0] : 100.100.100.100
  if_index : 31 (utun4)
  flags    : Supplemental, Request A records, Request AAAA records
  reach    : 0x00000003 (Reachable,Transient Connection)
  order    : 101401

resolver #7
  domain   : local
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300000

resolver #8
  domain   : 254.169.in-addr.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300200

resolver #9
  domain   : 8.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300400

resolver #10
  domain   : 9.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300600

resolver #11
  domain   : a.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300800

resolver #12
  domain   : b.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 301000

DNS configuration (for scoped queries)

resolver #1
  search domain[0] : local
  nameserver[0] : 192.168.0.2
  if_index : 12 (en0)
  flags    : Scoped, Request A records
  reach    : 0x00020002 (Reachable,Directly Reachable Address)

resolver #2
  search domain[0] : xxx.ts.net
  search domain[1] : xxxtailscale.net

```

Bug report created when everything was working fine again. I will provide more details if I can reproduce it.

### OS

macOS

### OS version

OSX 14.5

### Tailscale version

1.68.1

### Bug report

BUG-b41f96685e4018e3dbfeaa3e56071dbb737c1eac2937f4a004f1b70c252b00e4-20240626074844Z-e5089cdb8bb4e754

**Comment by agottardo** [comment]
@barnstar is currently working on improvements in this area that will ship in 1.70.

**Comment by agottardo** [comment]
@shuther I suggest trying the current unstable builds, as there are some fixes in there to mitigate this that won't ship until 1.70 is out next month. I would also recommend moving away from the App Store release, and switching to our [Standalone](https://tailscale.com/kb/1065/macos-variants) distribution when you get a chance.

**Comment by venkytv** [comment]
I suppose the fix for this did not make int into 1.70.0? I have 1.70.0 and this is still happening.

**Comment by agottardo** [comment]
We believe this is fixed in v1.74.1. If you're still experiencing issues, please tag me here and attach a bug report identifier.

**Comment by venkytv** [comment]
@agottardo  v1.74.1 was not released for macOS. I am on v1.76.0 and I still see this happening. The amount of time it takes to fix itself *might* have reduced perhaps, but it still takes about a minute to start resolving DNS queries.
Bug report identifier: BUG-94a3a56f4a2528f5af0a64c815542e3f4c0396694e4bce77429ce54dfd6724cb-20241017065129Z-fef6ce1ca76c3b1e
